src/cfnlint/schemaManager.py

```diff
@@ -0,0 +1,96 @@
+"""
+Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: MIT-0
+"""
+
+import json
+import logging
+import os
+import platform
+
+import botocore.exceptions
+import boto3
+from cfnlint.template import Template
+
+LOGGER = logging.getLogger(__name__)
+
+
+class SchemaManager(object):
+    """Download schemas if necessary and validate modules"""
+    def __init__(
+            self, filename, template, regions):
+        self.filename = filename
+        self.regions = regions
+        self.template = Template(filename, template, regions)
+
+    # Check if the appropriate folder already exists
+    def check_folders(self, client, name, registry_type):
+        account_id = client.get_caller_identity().get('Account')
+        username = None
+        path_split = os.getcwd().split('/')
+        try:
+            index = path_split.index('Users')
+            username = path_split[index + 1]
+        except ValueError:
+            raise ValueError
+
+        is_windows = platform.system() == 'win32'
+
+        for region in self.regions:
+            if is_windows:
+                path = 'C:/Users/{0}/AppData/cloudformation/{1}/{2}/{3}'.format(username, account_id, region, name)
+            else:
+                path = '/Users/{0}/.cloudformation/{1}/{2}/{3}'.format(username, account_id, region, name)
```

Comment:
nitpick - you can extract them into separate method.

Reply:
Done